libsyclinterface/source/dpctl_sycl_device_interface.cpp

```diff
@@ -849,3 +849,57 @@ DPCTLDevice_GetSubGroupSizes(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     }
     return sizes;
 }
+
+__dpctl_give DPCTLDeviceVectorRef
+DPCTLDevice_GetComponentDevices(__dpctl_keep const DPCTLSyclDeviceRef DRef)
+{
+    using vecTy = std::vector<DPCTLSyclDeviceRef>;
+    vecTy *ComponentDevicesVectorPtr = nullptr;
+    if (DRef) {
```

Comment:
In some functions we are checking for `XRef` to be non `nullptr`. (https://github.com/IntelPython/dpctl/pull/1993/files#diff-93f8a41da82dd5dd2ae6b05e019d8104c87c244e6077d5a819625a2fc5e3a2f1R858)

In others we are checking that result of `auto X = unwrap<...>(XRef)` is non `nullptr`. (https://github.com/IntelPython/dpctl/pull/1993/files#diff-93f8a41da82dd5dd2ae6b05e019d8104c87c244e6077d5a819625a2fc5e3a2f1R883, https://github.com/IntelPython/dpctl/pull/1993/files#diff-2cee49fe41f858d191076f530e6c6ce6ca841be844b8bca525509ed3deddbbaeR324)

Is that intentionally?
Are these checks interchangeable?
Can we `unwrap<...>(XRef)` if `XRef` is nullptr?
Can result of `unwrap<...>(XRef)` be nullptr if `XRef` is not nullptr?

Reply:
They are treated as more or less interchangeable throughout the file.

As to if we can `unwrap` a nullptr, yes, there are a few tests which more or less do this throughout the libsyclinterface tests.

And the last question, I can't think of such a case. Given it's treated as interchangeable: probably not.